pyanaconda/modules/common/task/progressible.py

```diff
@@ -0,0 +1,95 @@
+#
+# Copyright (C) 2018 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+from abc import ABC, abstractmethod
+from threading import Lock
+
+from pyanaconda.core.signal import Signal
+from pyanaconda.core.async_utils import async_action_nowait
+
+__all__ = ['Progressible']
+
+
+class Progressible(ABC):
```

Comment:
It is a real word :-) https://en.wiktionary.org/wiki/progressible

Reply:
I know it is but I think it is not describing purpose of this class.

Reply:
What about Reportable?

Reply:
I understand that you are trying to use the Runnable name convention but it sounds weird to me and I'm not sure that it is obvious what the purpose of this class. I don't think that Reportable is the correct way -- what is reported Errors, Progress...

The Progressible variant is slightly better. Still something like ProgressReportingMixin is more obvious to me. However I know this is subjective so I won't block this PR because of this.

Reply:
Ok, I have changed it to ProgressReporter, that should be more specific.